providers/default.rb

```diff
@@ -636,8 +636,10 @@ def execute_install_script(install_script)
                   else
                     '/etc/init.d/chef-client start'
                   end
+      env = {}
       unless node['chef_client']['chef_license'].nil?
-        start_cmd = "env CHEF_LICENSE=\"#{node['chef_client']['chef_license']}\" " + start_cmd
+        env['CHEF_LICENSE'] = node['chef_client']['chef_license']
+        fork { Kernel.exec(env, 'chef-client', '-z') }
```

Comment:
@dheerajd-msys can you take another try at this? Sounds like we've got a lot of impacted users. Let us know if you have any questions about Lamont's feedback.

Reply:
Yes, we are working on it @OBrienCommaJosh

Reply:
I'm not a fan of directly invoking the code since its an internal API and subject to change. My recommendation is to continue setting the environment variable and run `#{chef_install_dir}/bin/chef-apply -e "exit 0"`. This should be side-effect free.

Another option is to directly create the `~/.chef/accepted_licenses` files but I would recommend invoking the license acceptance flow via `chef-apply` before doing this.

Reply:
We are updating `#{chef_install_dir}/bin/chef-apply -e "exit 0"` after setting environment variable at here https://github.com/chef-cookbooks/chef_client_updater/blob/88ab0a608844a8a8883d8c7088d6166dce354c6e/providers/default.rb#L642 but this is not working as expected neither with `fork` nor without `fork`.

Reply:
Updated now with working chef-apply changes.

Reply:
@tyler-ball that seems like a better compromise, but does feel like we should have some way of only doing license acceptance without any extra machinery involved.